### ISSUE TITLE
Change filters attributes to be more loop specific

### ISSUE THREAD

**Issue by vladkucherov** [siteorigin/siteorigin-panels#17]
Hi guys,

First I want to say that the Page Builder is awesome! Really knocks out all your competitors!
For my project I had to modify the classes the Page Builder generates, and for that I used your filters.

in this particular issue, I am trying to add a class that says how much columns I have using:

```
siteorigin_panels_row_cell_classes
```

the problem is that I the whole panels data object instead the on in the loop.

```
$panels_data
```

_siteorigin-panels/siteorigin-panels.php_ **Line:** 850

Please change this to:

``` php
$panels_data['grids'][$gi]
```

Thanks,
Vlad.

**Comment by gregpriday** [comment]
Hi Vlad

Thanks for the suggestion. You're correct, there should be more data for that filter.

I'm having a proper look at all the filters in `siteorigin_panels_render`, and I can see there are a few inconsistencies with what data is being passed to each one. I'm going to think through this properly to try ensure we don't introduce any more inconsistencies.

Right now, I'm thinking that if we introduce a `['current']` key to the `$panels_data` array that includes the current row and cell indexes, that should be enough to ensure backwards comptibility and a good level of consistency.

I'll experiment with this idea though.

**Comment by vladkucherov** [comment]
That would be a solution, but it may take more resources then needed. the panels data object is already pretty loaded with data... maybe you'll run 2 filters and call this one deprecated?

**Comment by AlexGStapleton** [comment]
It appears this issue was resolved sometime around the 2.4 update. [siteorigin_panels_row_cell_classes](https://github.com/siteorigin/siteorigin-panels/blob/2.10.6/inc/renderer.php#L788) filter now includes an additional argument, `$cell`. This is equivalent to `$panels_data['grids'][$gi]`.
